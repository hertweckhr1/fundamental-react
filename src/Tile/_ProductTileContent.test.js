import { mount } from 'enzyme';
import ProductTile from './ProductTile';
import ProductTileContent from './_ProductTileContent.js';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ProductTileContent />', () => {
    const productTileContent = (
        <ProductTileContent title='Tile Title'>
            <p>Tile Description</p>
        </ProductTileContent>
    );

    test('create product tile content component', () => {
        // product media tile
        let component = renderer.create(productTileContent);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ProductTileContent component', () => {
            const element = mount(<ProductTile.Content data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the ProductTileContent component\'s h2 element', () => {
            const element = mount(<ProductTile.Content titleProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('h2').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
