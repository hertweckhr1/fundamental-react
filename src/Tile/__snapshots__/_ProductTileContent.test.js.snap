// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileContent /> create product tile content component 1`] = `
<div
  className="fd-product-tile__content"
>
  <h2
    className="fd-product-tile__title"
  >
    Tile Title
  </h2>
  <p>
    Tile Description
  </p>
</div>
`;
